packages/blocks/src/api/templates.js

```diff
@@ -115,6 +115,21 @@ export function synchronizeBlocksWithTemplate( blocks = [], template ) {
 					normalizedAttributes
 				);
 
+			const ignoredHookedBlocks = Object.values(
+				getHookedBlocks( blockName )
+			).flat();
+
+			if ( ignoredHookedBlocks.length ) {
+				const { metadata, ...otherAttributes } = blockAttributes;
+				blockAttributes = {
+					metadata: {
+						ignoredHookedBlocks,
+						...blockAttributes.metadata,
```

Comment:
Should it be the other way around in case the metadata has ignored hooked blocks listed? Not sure if that would ever happen:
```suggestion
						...blockAttributes.metadata,
						ignoredHookedBlocks,
```

Reply:
Theoretically, it could happen; these blocks and attributes ultimately come from the `template` that a container blocks sets for the default state of its inner blocks. (Although it's probably not advisable for block authors to manually set `metadata.ignoredHookedBlocks`.

But yeah, it might make sense to flip the order; or maybe to even concatenate both arrays.

Reply:
https://github.com/WordPress/gutenberg/pull/58553/commits/ea6fc048c9fd92c43425ac494053c4c2b9f8d254